### DISCUSSION TITLE
Does someone uses detekt Project Metrics?

### DISCUSSION THREAD

**Discussion by BraisGabin** [detekt/detekt#6452] [General]
Detekt 1.X not only finds issues in the code. It also give some `ProjectMetrics` about the project. Those values are displayed in 4 different reports:

<details><summary>the html report</summary><img src="https://github.com/detekt/detekt/assets/721244/6cb1fcc7-537c-4a11-8adb-22995f940bd3"/></details>
<details><summary>the md report</summary><img src="https://github.com/detekt/detekt/assets/721244/e2e3449c-ed0e-4cbb-8330-70fe20ba8686" /></details>
<details><summary>The console `ProjectStatisticsReport`</summary><img src="https://github.com/detekt/detekt/assets/721244/becd50ef-f425-4c29-b99f-b079e7f6ae13" /></details>
<details><summary>The console `ComplexityReport`</summary><img src="https://github.com/detekt/detekt/assets/721244/66095790-04b6-4327-b362-abf9fad91c67" /></details>

In the last ~4 years I didn't see a single issue or comment on slack reporting a bug, asking for how it works or anything similar in relation of those reports. That could means 2 things:
- It is feature complete
- No one cares about it

We talk about this topic 3 years ago at #2680 (search for the keyword `metric` because there are a lot of topics there).

I would like to remove it to reduce our public API and reduce the things that our core needs to handle. We want to rework our code so moving things out would be beneficial. I create the PR #6457 to show how much code we could remove if we drop this feature.

The question is. Does someone uses these reports for something? If so, could you explain your use case?

**Comment by schalkms** [comment]
Yes, we do.

I don't think there's much to be added to the metrics, so it's basically feature complete. What I remember from the last two years:
- We introduced cognitive complexity.
- We added Kotlin's scope functions to detekt's McCabe complexity implementation.
- `ComplexMethod` rule uses the complexity implementation.

Just search for **McCabe** and you find some issues, for instance.
https://github.com/detekt/detekt/issues?q=is%3Aissue+McCabe

**Reply by BraisGabin** [reply]
Hi @schalkms my initial description of this discussion was really missleading and it doesn't contain too much context. I rewrote it from scratch, could you re-read and see if your comment stands? I'm not talking about removing our rules about complexity nor the algorithms to calculate it. Those are great! I just "want" to remove the `ProjectMetric`s because I think that they provide really few value.

And, similar to `formatting`, we could even move it to a first party plugin. This will give us a bit more of work (extract it and then maintain) but I think that it would be less that maintain it as it is.

**Reply by schalkms** [reply]
My comment pretty much stands. I know that you don't want to remove the rules using metrics.

Anyways, I don't see the benefit of removing `ProjectMetric` in terms of value/effort, because I don't think it causes too much maintenance.
